units/volume_expiration_check.go

```diff
@@ -0,0 +1,89 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	volumeExpirationCheckName = "volume-expiration-check"
+)
+
+func init() {
+	registry.AddJobType(volumeExpirationCheckName,
+		func() amboy.Job { return makeVolumeExpirationCheckJob() })
+}
+
+type volumeExpirationCheckJob struct {
+	job.Base
+	VolumeID string `bson:"volume_id" json:"volume_id" yaml:"volume_id"`
+	Provider string `bson:"provider" json:"provider" yaml:"provider"`
+
+	volume *host.Volume
+	env    evergreen.Environment
+}
+
+func makeVolumeExpirationCheckJob() *volumeExpirationCheckJob {
+	j := &volumeExpirationCheckJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    volumeExpirationCheckName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func NewVolumeExpirationCheckJob(ts string, v *host.Volume) amboy.Job {
+	j := makeVolumeExpirationCheckJob()
+	j.SetID(fmt.Sprintf("%s.%s", volumeExpirationCheckName, ts))
+	j.VolumeID = v.ID
+	j.Provider = evergreen.ProviderNameEc2OnDemand
```

Comment:
should we just do the job per host? why is it right to set the provider at this level? 

should be be stashing the volume here? other wise line 65 is always true.

Reply:
> should we just do the job per host?

Could you elaborate?

Reply:
> why is it right to set the provider at this level?

I moved it to a parameter

> should be be stashing the volume here?

done